rfc-139-enable-http3.md

```diff
@@ -0,0 +1,135 @@
+
+Each TCP packet now includes a set of "frames" that contains additional meta information about each stream. Each asset is downloaded over a separate stream and then reconstructed by the browser. This allows multiple assets to be downloaded at the same time over a single TCP connection, removing the HOL blocking at the application (HTTP) layer. An example of this setup can be seen in the diagram below: 
+
+![An example of HTTP/2 frames within TCP packets.](rfc-139/5_H2_2files_multiplexed.png)
+
+Source: ["Head-of-Line Blocking in QUIC and HTTP/3: The Details"](https://calendar.perfplanet.com/2020/head-of-line-blocking-in-quic-and-http-3-the-details/) by [Robin Marx](https://twitter.com/programmingart).
+
+Unfortunately this didn't solve all the HOL blocking issues. Since HTTP/2 is built upon TCP, and TCP is a protocol that offers the reliable transmission of packets. TCPs job is to make sure that the packets leaving one device arrive at another device in the exact same order and without any missing packets. When packet loss occurs, this is when we again see HOL blocking. When a packet of data is lost over the network, this then holds up the whole (single) TCP connection. Packets behind the missing packet can't be processed until retransmission occurs.
+
+This is where we are currently at with GOV.UK. The vast majority of users on strong, stable connection will have seen improved performance from enabling HTTP/2. Unfortunately, some users may have seen performance drop if they use a connection with high packet loss. This is due to the HOL blocking that now exists at the network layer. This is a primary issue HTTP/3 sets out to solve.
+
+### HTTP/3
+To solve the HOL blocking issue at the network layer, HTTP/3 uses a completely new transport layer protocol called QUIC. QUIC initially stood for "Quick UDP Internet Connections", but the IETF now don't use it as an acronym. It is simply the name of the protocol. QUIC is built upon User Datagram Protocol (UDP). Since QUIC uses UDP, it isn't bound by the restrictions that TCP imposes. So it has now been designed so that the network layer now understands the the concept of independent streams, which was restricted to the application layer under HTTP/2:
+
+![Difference between TCP and QUIC when downloading assets.](rfc-139/QUIC-illustration-final.jpg)
+
+In the illustration above, if packet 2 is lost (using HTTP/2), packets 3-8 are blocked until retransmission occurs. Using QUIC if packet 2 is lost only packets 1 and 3 are blocked since they exist in the same stream. The other streams can progress as normal since they are fully independent. QUIC doesn't require in-order packet transmission across the whole connection, but it does still retain ordering within each resource stream.
```

Comment:
In the illustration and when using QUIC, if packet 2 is lost, would it be both packets 1 and 3 which are blocked (if transmitted out of order), or only packet 3 which is lost (if transmitted in order)? Apologies if I've missed this.

Reply:
So both packet 1 and 3 would be blocked. Packet 1 and 3 would exist in a buffer on the client waiting for packet 2 to arrive (under QUIC). The ordering of the packets is retained within independent streams under QUIC. All packets need to be present before the whole stream (and therefor the asset) can be processed by the client.